logging/logging.go

```diff
@@ -0,0 +1,60 @@
+package logging
+
+import (
+	"log"
+
+	raven "github.com/getsentry/raven-go"
+)
+
+// Logger describes the interface for the loggers in this package.
+// Use Info for informational logs (including errors that are silenced).
+// Use Error for errors.
+type Logger interface {
```

Comment:
I was surprised to see this- it's no secret that logging in golang is a mess, but I think it's worth going with something that already exists.

The [prometheus logger](github.com/prometheus/common) is what you're looking for here, I think. It wraps logrus and presents a consistent logging interface. The benefit here should be clear: you can ditch this logging package entirely, use the available Sentry hook, get structured logging for free and make it really easy to add other hooks in future should that become necessary.

What're your thoughts here?

```golang
package main

import (
	"github.com/prometheus/common/log"
)

func main() {
	log.With("some", "context").Infoln("hello world")
}
```

```
λ ~> go run main.go
INFO[0000] hello world                                   some=context source="main.go:8"
```

Reply:
Yeah, I mean it shouldn’t really be that surprising that I didn’t use a logger interface from a somewhat arbitrary go project- it’s not the most discoverable thing in the world.

The Prometheus logger is pretty much identical to this one but with a load of methods that aren’t needed right now. A Prometheus logger that wraps logrus isn’t really any different from this logger wrapping logrus.

Reply:
The prometheus logger is one of the current attempts to standardise an interface for golang logging, so I think it's not too much of a cost to bring in those other methods. I feel like it's quite inevitable you'll want to use extra levels like debug etc later, and that would require a subsequent change to the logger package. 

This wrapper also doesn't support structured logging, and you don't have the community of hooks that logrus does. I don't mind if we implement our own version of the prometheus logger here, but I'd really like this to go through logrus so we can benefit from the community hooks, one of which being sentry. 

Can we also add the source line annotation that the prometheus logger adds please? It makes debugging from log lines so much easier.

Reply:
Chatted with @hmac out-of-band, we've decided to use the prometheus logger given it's community momentum and logrus integration out the box. Going to add a commit to this PR with this change and pass it back for review.